### ISSUE TITLE
API Umbrella Community Portal

### ISSUE THREAD

**Issue by brylie** [NREL/api-umbrella#122]
Set-up a community portal for API Umbrella users, developers, maintainers, etc.

User Story
------------
As an API Umbrella user
I would like a community portal
so that I can connect with other API Umbrella users

**Comment by brylie** [comment]
Some possible community portals include:

* [Anahita](http://www.getanahita.com/)
* [Elgg](http://elgg.org/)
* [Discourse](http://www.discourse.org/)

**Comment by perfaram** [comment]
I've got a not-fully used server at DigitalOcean. (or, if the portal uses pure PHP, then I can set it up on my server at OVH)

**Comment by brylie** [comment]
@perfaram great! Lets figure out the most important features of a community platform, so that we can choose a solution.

**Comment by perfaram** [comment]
Hmmmm, don't know...

**Comment by brylie** [comment]
I would like to propose we use the Discourse platform, as it is quite user friendly and feature rich.

**Comment by brylie** [comment]
@GUI any thoughts?

**Comment by GUI** [comment]
Great idea, thanks!

What would your thoughts be on a Google Group? I like Discourse too, and it offers some additional features that are nice, but I'm also a fan of relying on hosted services that we don't have to host ourselves, keep up to date, apply security patches to, etc.

Are there any particular features of Discourse you'd like over Google Groups? If so, we could certainly look at getting Discourse setup somewhere. But if you think Google Groups might work, it would certainly be easy to set something up immediately and give it a spin.

**Comment by perfaram** [comment]
Personally, I don't trust much Google for keeping their services working in the long-run. Look at what they're doing to Google Code... Sure, *we* don't have to care about keeping it up-to-date and safe, but what if Google just closes it ?

**Comment by brylie** [comment]
Perceval speaks my mind.

On 23 March 2015 14:35:37 EET, Perceval F <notifications@github.com> wrote:
>Personally, I don't trust much Google for keeping their services
>working in the long-run. Look at what they're doing to Google Code...
>Sure, *we* don't have to care about keeping it up-to-date and safe, but
>what if Google just closes it ?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/NREL/api-umbrella/issues/122#issuecomment-84974338

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by GUI** [comment]
Yup, those are definitely valid concerns. Although, I think even in the event Google decides to close a product, they usually give ample warning and opportunity to migrate to other platforms. So I guess I might personally still be inclined to give Google Groups a shot since it's so easy to get setup. If we later decide we need additional functionality, or in the event Google does shutter their Groups service, we could always migrate elsewhere at that point. But for now, it seems like Google Groups might be a quick, easy, and free first option. If that sounds reasonable to you all, I'd be happy to get the Google Group setup. Or if you'd all definitely prefer Discourse, I can also look into finding hosting for that.

Thanks for the feedback!

**Comment by brylie** [comment]
I prefer Discourse, as it is open-source. Bitnami has a [Discourse installer](https://bitnami.com/stack/discourse), which may be useful. What are some options for hosting?

**Comment by brylie** [comment]
I just chatted with Chandra, and we can offer a free VM with 2G RAM with 40GB HDD to support this initiative.

**Comment by perfaram** [comment]
Seems good !

**Comment by brylie** [comment]
@GUI we would like to use the subdomain community.apiumbrella.io for the community portal. Will you please create the sub domain? We will set up the server and tell you the IP, so that you can update the DNS settings.

**Comment by kyyberi** [comment]
+1

**Comment by GUI** [comment]
Thanks @brylie! I don't yet have DNS access for apiumbrella.io, but I'm actually in the midst of getting that transferred. So hopefully later this week I'll be able to setup the community.apiumbrella.io subdomain for you. I'll keep you posted.

**Comment by brylie** [comment]
@GUI what is the status of the apiumbrella.io subdomain? We are working to set up a VM and install the community platform, and will need a domain name to complete the task.

**Comment by GUI** [comment]
The domain name transfer for apiumbrella.io started yesterday. It sounds like it typically takes another 5-7 days before the transfer process completes. But assuming we don't hit any snags, then we should be able to easily setup this subdomain. Just let me know what IP or hostname you'd like us to point things at.

**Comment by ccsr** [comment]
BTW where is it getting transferred?

**Comment by GUI** [comment]
@ccsr: The domain is being transferred to an NREL-managed account. We can certainly discuss migrating it to some place else in the future if that makes more sense for the community, but for now it seemed like an NREL account was the best fit, so it wasn't under anyone's personal account.

**Comment by brylie** [comment]
Yep, all we need for the community platform is a sub-domain (e.g. community.apiumbrella.io) pointing to a specific IP. The management can be left with NREL.

**Comment by brylie** [comment]
@GUI based on recent discussion, the IP address seems to be 109.204.237.172. Please point the chosen sub-domain to 109.204.237.172.

**Comment by ccsr** [comment]
@GUI we need to verify also the domain to use mail gun. We can use our account but domain needs to be verified.

**Comment by GUI** [comment]
@ccsr, @brylie: I just got confirmation we've switched the NS records over to the new location for the domain. The NS record TTLs were 24 hours, so unfortunately it may take a while for this to fully propagate to people who had previously accessed apiumbrella.io and is being cached by their DNS servers. But new people accessing the domain should likely land on the correct domain immediately: https://community.apiumbrella.io I also have access to easily tweak the DNS now, so if you do need to setup any further records for mailgun, just let me know.

**Comment by brylie** [comment]
Community.apiumbrella.io works for me! Great work :-)

On 21 September 2015 19:39:59 EEST, Nick Muerdter <notifications@github.com> wrote:
>@ccsr, @brylie: I just got confirmation we've switched the NS records
>over to the new location for the domain. The NS record TTLs were 24
>hours, so unfortunately it may take a while for this to fully propagate
>to people who had previously accessed apiumbrella.io and is being
>cached by their DNS servers. But new people accessing the domain should
>likely land on the correct domain immediately:
>https://community.apiumbrella.io I also have access to easily tweak the
>DNS now, so if you do need to setup any further records for mailgun,
>just let me know.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/NREL/api-umbrella/issues/122#issuecomment-142037561

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by kyyberi** [comment]
Excellent :)

**Comment by brylie** [comment]
Hey @kyyberi would you please help us by starting some discussions on the new community platform?

**Comment by kyyberi** [comment]
I'll try to find time for that

**Comment by jykae** [comment]
I can't login with Github to Discourse after community integration: http://community.apinf.io/auth/failure?message=redirect_uri_mismatch&strategy=github

"Sorry, there was an error authorizing your account. Perhaps you did not approve authorization?"

Github auth request redirects to wrong URL.

**Comment by ccsr** [comment]
@jykae I just signed in and it works

**Comment by jykae** [comment]
Nice to hear @ccsr But I didn't get any approval like the error message says. Tried to remove application from Github and tried again. Same error, maybe something cached in browser memory..

**Comment by jykae** [comment]
I think I found the problem. 
community.apinf.io doesn't use HTTPS. It should.
community.apiumbrella.io uses HTTPS. When I login with Github from community.apiumbrella.io redirect fails. But Github login works when using the same domain :) Cross-domain redirections are also a bit insecure, when not using HTTPS.
